src/sources/lolalytics.js

```diff
@@ -0,0 +1,304 @@
+          $c(row).find('td.popularity.summarystats').each(function(z, pickPercent) {
+            build_section.json[z].pick = parseFloat($c(pickPercent).text());
+          }); //  end find popularity
+        }
+      } //  end else (not items)
+    }); //  end each row
+    build_sections.push(build_section);
+  }); //  end each summarytitle
+
+  return build_sections;
+}
+
+function parseChampionPage(cheerio_data, position, champion_name) {
+  const build_sections = parseItemData(cheerio_data);
+  const skills = parseSkillsData(cheerio_data);
+  const parsed_data = {
+    build_sections: build_sections,
+    skills: skills
+  };
+  const builds = [];
+
+  if (store.get('settings').splititems) {
+    builds.push(createRiotJson(parsed_data, position, champion_name, 'Win'));
+    builds.push(createRiotJson(parsed_data, position, champion_name, 'Pick'));
+  } else {
+    builds.push(createRiotJson(parsed_data, position, champion_name));
+  }
+  return builds;
+}
+
+function parseChampion(request_params, cheerio_data) {
+  const $c = cheerio_data;
+  const champion_name = request_params.champion_name;
+  const position = request_params.position;
+  let formatted_builds = [];
+
+  const positions = $c('.lanebox > div.lanetitle');
+  let position_names = [];
+  let current_position = null;
+  positions.each(function(i, pos) {
+    $c(pos).children().remove(); // Remove the x.xx% after the lane name, it's a child in the div
+    position_names.push($c(pos).text());
+    if ($c(pos).parent().hasClass('selected'))
+      current_position = $c(pos).text();
+  });
+  position_names = R.filter(role => (role !== position && role !== 'All Lanes'), position_names); //  Remove if exists
+
+  if (position_names.length === 1 || position) { // Single position champ or we were passed a position
+    formatted_builds = R.concat(formatted_builds, parseChampionPage(cheerio_data, current_position, champion_name));
+  }
+
+  if (position_names.length > 1 && !position) { //  More positions available and we werent passed one
+    let extra_requests = [];
+    extra_requests = R.map(position => ({champion_name, position}), position_names);
+    return Promise.resolve(extra_requests)
+    .map(request_params => getChampionPage(request_params)) // eslint-disable-line no-use-before-define
+    .then(R.flatten)
+    .then(R.concat(formatted_builds));
+  }
+
+  return formatted_builds;
+}
+
+function getChampionPage(request_params) {
+  const { champion_name, position } = request_params;
+  let url = `http://current.lolalytics.com/champion/${champion_name}/`;
+  if (position)
+    url += position + '/';
+  else {
+    cl(`${T.t('processing')} Lolalytics: ${T.t(champion_name)}`);
+  }
+
+  function markUndefined() {
+    store.push('undefined_builds', {
+      source: source_info.name,
+      champion_name,
+      position: request_params.position || 'All'
+    });
+    return;
+  }
+
+  return request(url).then(html => {
+    console.timeEnd(champion_name + position);
```

Comment:
Yeah this was when my tests were timing out before I realized I was missing an extra '/' at the end of my nock urls